extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php

```diff
@@ -72,35 +66,35 @@ public function executeIndex() {
 		wfProfileOut( __METHOD__ );
 	}
 
-	public function executeThemeTab() {
+	public function themeTab() {
 
 	}
 
-	public function executeCustomizeTab() {
+	public function customizeTab() {
 
 	}
 
-	public function executeWordmarkTab() {
-		$this->faviconUrl = Wikia::getFaviconFullUrl();
+	public function wordmarkTab() {
+		$this->response->setData( [
+			'faviconUrl' => Wikia::getFaviconFullUrl(),
+			'token' => $this->wg->User->getEditToken(),
+		] );
 	}
 
-	public function executePicker() {
-
+	public function picker() {
+		// SUS-797: Add edit token for background image upload
+		$this->response->setVal( 'token', $this->wg->User->getEditToken() );
 	}
 
-	public function executePreview() {
-		global $wgEnablePortableInfoboxEuropaTheme;
-
-		$this->infoboxTheme = $wgEnablePortableInfoboxEuropaTheme ? 'theme="europa"' : '';
+	public function preview() {
+		$this->infoboxTheme = $this->wg->EnablePortableInfoboxEuropaTheme ? 'theme="europa"' : '';
 	}
 
 	/**
 	 *
 	 * @author Federico "Lox" Lucignano
 	 */
 	private function getUploadErrorMessage( $status ) {
-		global $wgFileExtensions, $wgLang;
-
 		switch( $status[ 'status' ] ) {
 			case UploadBackgroundFromFile::FILESIZE_ERROR:
 				$msg = wfMsgHtml( 'themedesigner-size-error' );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)